### PR TITLE
Samples: use more common images

### LINKED ISSUE #28071: Replace governmentpaas/curl-ssl with curlimages/curl in samples
Just to use a more official image

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Xiang Dai <long0dai@foxmail.com>

Fix #28071

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[X] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

/release-note-none

### CODE REVIEW

File: samples/https/nginx-app.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ spec:
           name: nginxconfigmap
       containers:
       - name: nginxhttps
-        image: ymqytw/nginxhttps:1.5
+        image: nginx:1.14.2
```
Comment (howardjohn) [inline samples/https/nginx-app.yaml:36]: I think we can actually remove this whole file, its not used in the docs and clearly super old  (see "ReplicationController")

**Review by howardjohn — APPROVED** [review]

File: samples/https/nginx-app.yaml (hunk 1 above)
Comment (daixiang0) [reply samples/https/nginx-app.yaml:36]: Get.

File: samples/https/nginx-app.yaml [hunk 2]
```diff
@@ -1,44 +0,0 @@
-apiVersion: v1
```
Comment (howardjohn) [inline samples/https/nginx-app.yaml:1]: should we remove the whole folder? The other file was just a config file for this sample

**Review by howardjohn — APPROVED** [review]

File: samples/https/nginx-app.yaml (hunk 2 above)
Comment (ericvn) [reply samples/https/nginx-app.yaml:1]: I agree that we should remove the folder if we are removing the file. Looking at istio.io, these files are called out most recently in the 1.5 archive (when referring to the English docs (the Chinese docs which are behind have it in their current docs. (discussion to not display translated docs more than a yr old in the future))).

**Comment by ericvn** [comment]
@frankbu Comments on removing samples called out in archived versions of the docs. If the archived version of the docs is no longer supported, is that sufficient or should we go back further? I'm not sure if the link checker is run against the archived pages and would need to verify. Might require updating links to specific branches of istio/istio as well.

**Comment by frankbu** [comment]
@ericvn, the links from docs to github files in istio/istio are branch specific, so we should be OK.

File: samples/https/nginx-app.yaml (hunk 2 above)
Comment (frankbu) [reply samples/https/nginx-app.yaml:1]: The new translation model we're talking about moving to will automatically redirect to the english version of any doc if the translated version is out of date relative to the english one.

Comment (daixiang0) [reply samples/https/nginx-app.yaml:1]: Remove them.

**Review by ericvn — APPROVED** [review]

**Comment by ericvn** [comment]
@daixiang0 If you select the `Details` to the failing test, it will bring up the test. On that page is a link to the build logs.  From the build logs:
```
Missing release notes and missing "release-notes-none" label.
If this pull request contains user facing changes, please create a release note based on the template: https://github.com/istio/istio/blob/master/releasenotes/template.yaml by going here: https://github.com/daixiang0/istio/new/curl-image/releasenotes/notes
Release notes documentation can be found here: https://github.com/istio/istio/tree/master/releasenotes
If this pull request has no user facing changes, please add the release-notes-none label to the pull request. Note that the test will have to be manually retriggered after adding the label.
```

**Comment by daixiang0** [comment]
@ericvn hi, i add `/release-note-none` as comment while not help :(

**Pull request merged**